### ISSUE TITLE
Can't build on Mac OS X 10.12

### ISSUE THREAD

**Issue by jcbloch** [AdolfVonKleist/Phonetisaurus#25]
Build fails on Mac OS X 10.12 because some required openfst `*.so` files don't seem to exist anywhere (example: `libfstfar.so` is missing). I installed OpenfFST   for Kaldi, and Kaldi works fine. Same issue for 1.6.1, 1.6.2, and 1.6.3.  Setting `--with-openfst-libs` and `--with-openfst-includes` is no help (I can sym link to the required dyld files in the new dir structure just fine, e.g., `/usr/local/lib/fst/libfstfar.dylib -> /usr/local/lib/libfstfar.dylib`) Perhaps this is a openfst issue?  Thanks.

**Comment by AdolfVonKleist** [comment]
I have so far actually only tested against linux distros for the build script.  It might also be related to the ```LD_LIBRARY_PATH```, I have another outstanding request on that topic.  Can you clarify whether your issue is related to building/compiling or running the resulting utilities?  Or provide the raw error message?

**Comment by jcbloch** [comment]
Running `.configure ....` exits with `Can't find one or more extensions.  OpenFst should....`
regardless of what I set `with_openfst_libs` to or not.
Turns out `libfstfar.so` is not in either `usr/local/lib` or  `usr/local/lib/fst` (or anywhere else).
The main dyld files are in usr/local/lib, and the extensions in usr/local/lib/fst (hence the symlinks).
Also, on my system, `ngram-fst.so` (actually a symlink to `ngram-fst.0.so`) is here:  `usr/local/lib/fst`
Do these missing (?) files suggest to you a problem with openfst ?

**Comment by AdolfVonKleist** [comment]
Yes it suggests an issue with OpenFst.  I have not provisioned the configure script explicitly for OSX.  I can take a look at it probably this weekend.  Otherwise if you have a PR with a solution of your own that would also be welcome.

**Comment by jcbloch** [comment]
Thanks! FWIW, reverting to OpenFST 1.4, and more-or-less following  [this](https://github.com/AdolfVonKleist/Phonetisaurus/issues/8) I was able to install. I had to use an old commit though..  1f834c6ea02ddfeb3fdc9ec13b30dfedb20ad241
I also followed your instructions on a couple of clean linux list's and it worked fine.  I'm still looking for a OpenFST 1.6 Mac solution, but I won't have time to find it (myself) anytime soon...
